executor/start.sh

```
@@ -0,0 +1,274 @@
+
+# Check if port is already in use
+check_port() {
+    local port=$1
+    local name=$2
+    
+    if lsof -Pi :$port -sTCP:LISTEN -t >/dev/null 2>&1; then
+        echo -e "${YELLOW}Warning: Port $port ($name) is already in use${NC}"
+        echo ""
+        
+        # Check for suspended jobs
+        local suspended_jobs=$(jobs -l | grep -i "suspended" | grep "start.sh" || true)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)